### PR TITLE
Reach: fix amount handling to work with cents properly

### PR DESCRIPTION
**Pull request by Heavyblade**
## Summary:

Provides proper format handling on reach gateway to receive amount as cents on authorize/purchase and refund.

## Remote Test:

Finished in 105.613556 seconds.
28 tests, 70 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed

## Unit Tests:
Finished in 30.429736 seconds.
5425 tests, 77010 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed

## RuboCop:
756 files inspected, no offenses detected

### CODE REVIEW

**Review by edgarv09 — APPROVED** [review]
Well done! 🤖  🚀

File: lib/active_merchant/billing/gateways/reach.rb [hunk 1]
```diff
@@ -82,12 +78,13 @@ def scrub(transcript)
...
 
       def refund(amount, authorization, options = {})
+        currency = options[:currency] || currency(options[:amount])
         post = {
           request: {
             MerchantId: @options[:merchant_id],
             OrderId: authorization,
             ReferenceId: options[:order_id] || options[:reference_id],
-            Amount: amount
+            Amount: localized_amount(amount, currency)
```
Comment (javierpedrozaing) [inline lib/active_merchant/billing/gateways/reach.rb:87]: I think we can include it also for capture request.

**Review by javierpedrozaing — COMMENTED** [review]
LGTM, just I left a comment for your consideration.

**Review by naashton — CHANGES_REQUESTED** [review]
I'm not sure why, but `test_successful_refund_with_reference_id` and `test_successful_refund_with_order_id` remote tests fail for me. They both pass on master branch though.

**Review by naashton — APPROVED** [review]
🚢

**Pull request merged**